### PR TITLE
update a few more cases that need to use explicit resource id creation

### PR DESCRIPTION
**Pull request by m-nash**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs [hunk 1]
```diff
@@ -98,8 +98,11 @@ protected virtual void WriteCtors()
             _writer.WriteXmlDocumentationSummary($"Initializes a new instance of the <see cref = \"{TypeOfThis.Name}\"/> class.");
             _writer.WriteXmlDocumentationParameter("options", $"The client parameters to use in these operations.");
             _writer.WriteXmlDocumentationParameter("resource", $"The resource that is the target of operations.");
+
+            var idPropString = _resourceData.IsIdString() ? "new ResourceIdentifier(resource.Id)" : "resource.Id";
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs:102]: Is it guaranteed that parameter always called `resource`?

File: src/AutoRest.CSharp/Mgmt/Output/ResourceData.cs [hunk 2]
```diff
@@ -22,5 +23,14 @@ protected string CreateDescription(string clientPrefix)
         {
             return $"A class representing the {clientPrefix} data model.";
         }
+
+        internal bool IsIdString()
+        {
+            var baseTypes = EnumerateHierarchy().TakeLast(2).ToArray();
+            var baseType = baseTypes.Length == 1 || baseTypes[1].Declaration.Name == "Object" ? baseTypes[0] : baseTypes[1];
+            var idProperty = baseType.Properties.Where(p => p.Declaration.Name == "Id").First();
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/Mgmt/Output/ResourceData.cs:31]: Is it possible that property name will change in future?

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs (hunk 1 above)
Comment (m-nash) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs:102]: yes

File: src/AutoRest.CSharp/Mgmt/Output/ResourceData.cs (hunk 2 above)
Comment (m-nash) [reply src/AutoRest.CSharp/Mgmt/Output/ResourceData.cs:31]: no this is an arm requirement

**Review by AlexanderSher — APPROVED** [review]

**Pull request merged**